[EDU-6046] Create Azion Lib docs in English (#1479)
aziontech/docs: src/content/docs/en/pages/devtools/azion-lib/usage/wasm-image-processor.mdx

---
title: WASM Image Processor library
description: >-
  Explore the `WASM Image Processor` library for efficient image processing using WebAssembly.
permalink: /documentation/products/azion-lib/wasm-image-processor/
meta_tags: >-
  Azion Lib, edge computing, edge applications, Azion products, Image Processor, WebAssembly
namespace: documentation_azion_lib_wasm_image_processor
---

> The Azion **WASM Image Processor Library** provides functions to process images using **WebAssembly**. This library allows you to load, resize, and retrieve images in various formats efficiently.

Reviewer: ```suggestion
The Azion **WASM Image Processor** library provides functions to process images using **WebAssembly**. This library allows you to load, resize, and retrieve images in various formats efficiently.
```

> 
> ## Usage

Reviewer 2: ```suggestion

---

## Usage
```

### loadImage

Loads an image from a URL or file path.

Example:

```typescript
import { loadImage } from 'azion/wasm-image-processor';
import type { WasmImage } from 'azion/wasm-image-processor';

const image: WasmImage = await loadImage('https://example.com/image.jpg');
```

> **Parameters:**

Reviewer: ```suggestion
**Parameters**:
```

| Parameter | Type | Description |
| --- | --- | --- |
| pathOrURL | string | The URL or file path of the image to load. |

> **Returns:**

Reviewer: ```suggestion
**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `Promise<WasmImage>` | A promise that resolves with a `WasmImage` instance. |

### resize

Resizes the loaded image.

Example:

```typescript
import { loadImage } from 'azion/wasm-image-processor';
import type { WasmImage } from 'azion/wasm-image-processor';

const image: WasmImage = await loadImage('https://example.com/image.jpg');
const resizedImage: WasmImage = image.resize(0.5, 0.5);
```

> **Parameters:**

Reviewer: ```suggestion
**Parameters**:
```

| Parameter | Type | Description |
| --- | --- | --- |
| `width` | `number` | The new width of the image. |
| `height` | `number` | The new height of the image. |

> | `usePercent?` | `boolean` | Whether to use percentages for resizing. Defaults to `true`. |

Reviewer: ```suggestion
| `usePercent?` | `boolean` | Whether to use percentages for resizing. Defaults are set to `true`. |
```

> | `usePercent?` | `boolean` | Whether to use percentages for resizing. Defaults to `true`. |

Reviewer 2: ```suggestion
| `usePercent?` | `boolean` | Whether to use percentages for resizing. Default value: `true`. |
```

> **Returns:**

Reviewer: ```suggestion
**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `WasmImage` | A new `WasmImage` instance with the resized image. |

### getImageResponse

Retrieves the processed image in the specified format.

Example:

```typescript
import { loadImage } from 'azion/wasm-image-processor';
import type { WasmImage, SupportedImageFormat } from 'azion/wasm-image-processor';

const image: WasmImage = await loadImage('https://example.com/image.jpg');
const imageResponse: Response = image.getImageResponse('jpeg' as SupportedImageFormat);
console.log(imageResponse);
```

> **Parameters:**

Reviewer: ```suggestion
**Parameters**:
```

| Parameter | Type | Description |
| --- | --- | --- |

> | `format` | `SupportedImageFormat` | The format of the image (`'jpeg'`, `'png'`, `'webp'`). |

Reviewer 2: ```suggestion
| `format` | `SupportedImageFormat` | The format of the image. Example: `'jpeg'`, `'png'`, `'webp'`. |
```

> | `quality?` | `number` | The quality of the image (for `'jpeg'`). Defaults to `100.0`. |

Reviewer: ```suggestion
| `quality?` | `number` | The quality of the image (for `'jpeg'`). Defaults are set to `100.0`. |
```

> | `quality?` | `number` | The quality of the image (for `'jpeg'`). Defaults to `100.0`. |

Reviewer 2: ```suggestion
| `quality?` | `number` | The quality of the image for `'jpeg'`. Default value: `100.0`. |
```

> **Returns:**

Reviewer: ```suggestion
**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `Response` | The response object containing the processed image. |

### clean

Cleans up the image data to free memory.

Example:

```typescript
import { loadImage } from 'azion/wasm-image-processor';
import type { WasmImage, SupportedImageFormat } from 'azion/wasm-image-processor';

const image: WasmImage = await loadImage('https://example.com/image.jpg');
image.clean();
```

> **Returns:**

Reviewer: ```suggestion
**Returns**:
```

| Return Type | Description |
|-------------|-------------|
| `void` | No return value. |

## Types

> These are the types used by the WASM Image Processor libreary and its methods:

Reviewer: ```suggestion
These are the types used by the WASM Image Processor library and its methods:
```

> These are the types used by the WASM Image Processor libreary and its methods:

Reviewer 2: ```suggestion
These are the types used by the **WASM Image Processor** library and its methods:
```

### WasmImage

An interface representing a wrapped PhotonImage with additional methods for image processing.

> **Properties:**

Reviewer: ```suggestion
**Properties**:
```

| Property | Description |
| --- | --- |
| `image` | The PhotonImage instance. |

> **Methods:**

Reviewer: ```suggestion
**Methods**:
```
